#+TITLE: snakemake-wrappers

An opinionated set of personal Snakemake wrappers.

- bedtools/cov-length :: Count the number of bases covered by a BED file
- [[bedtools/rename-chroms][bedtools/rename-chroms]] :: Rename BED file chromosomes
- [[bcftools/concat][bcftools/concat]] :: Concatenate VCF records
- [[bcftools/deduplicate][bcftools/deduplicate]] :: Deduplicate VCF records
- [[bcftools/filter][bcftools/filter]] :: Include or exclude variants from a VCF file
- [[bcftools/merge][bcftools/merge]] :: Merge samples of VCF files
- [[bcftools/normalize][bcftools/normalize]] :: Left-align and decompose multiallelic records
- [[bcftools/semijoin][bcftools/semijoin]] :: Semi join one VCF file against another
- [[bcftools/subset-samples][bcftools/subset-samples]] :: Subset one or more samples from a multi-sample VCF file, removing variants not in the subset
- [[epacts/make-gene-group-file][epacts/make-gene-group-file]] :: Create a gene-based marker group file for an ANNOVAR-annotated VCF
- [[epacts/qqplot][epacts/qqplot]] :: Generate a Q-Q plot from EPACTS results
- [[maf/concat][maf/concat]] :: Concatenate MAF files
- [[r-snprelate/pca][r-snprelate/pca]] :: Calculate PCA coordinates from a VCF file
- [[samtools/tabix][samtools/tabix]] :: Index VCF files
- [[vcf2maf/vcf2maf][vcf2maf/vcf2maf]] :: Convert a single-sample VCF file to MAF
